distribution/kernel/carbon-home/dbscripts/mssql.sql

```diff
@@ -684,3 +684,104 @@ CREATE TABLE UM_GROUP_UUID_DOMAIN_MAPPER (
+CREATE TABLE UM_ORG_GROUP(
+            UM_GROUP_ID VARCHAR(255) NOT NULL,
+            UM_GROUP_NAME VARCHAR(255) NOT NULL,
+            UM_TENANT_ID INTEGER DEFAULT 0,
+            PRIMARY KEY(UM_GROUP_ID)
+);
+
+IF NOT  EXISTS (SELECT * FROM SYS.OBJECTS WHERE OBJECT_ID = OBJECT_ID(N'[DBO].[UM_ORG_PERMISSION]') AND TYPE IN (N'U'))
+CREATE TABLE UM_ORG_PERMISSION(
+            UM_ID INTEGER IDENTITY(1,1) NOT NULL,
+            UM_RESOURCE_ID VARCHAR(255) NOT NULL,
+            UM_ACTION VARCHAR(255) NOT NULL,
+            UM_TENANT_ID INTEGER DEFAULT 0,
+            PRIMARY KEY(UM_ID)
+);
+
+IF NOT  EXISTS (SELECT * FROM SYS.OBJECTS WHERE OBJECT_ID = OBJECT_ID(N'[DBO].[UM_ORG_ROLE_USER]') AND TYPE IN (N'U'))
+CREATE TABLE UM_ORG_ROLE_USER (
+            UM_USER_ID VARCHAR(255) NOT NULL,
+            UM_ROLE_ID VARCHAR(255) NOT NULL,
+            CONSTRAINT FK_UM_ORG_ROLE_USER_UM_ORG_ROLE FOREIGN KEY (UM_ROLE_ID) REFERENCES UM_ORG_ROLE(UM_ROLE_ID) ON DELETE CASCADE
+);
+
+IF NOT  EXISTS (SELECT * FROM SYS.OBJECTS WHERE OBJECT_ID = OBJECT_ID(N'[DBO].[UM_ORG_ROLE_GROUP]') AND TYPE IN (N'U'))
+CREATE TABLE UM_ORG_ROLE_GROUP(
+            UM_GROUP_ID VARCHAR(255) NOT NULL,
+            UM_ROLE_ID VARCHAR(255) NOT NULL,
+            CONSTRAINT FK_UM_ORG_ROLE_GROUP_UM_ORG_ROLE FOREIGN KEY (UM_ROLE_ID) REFERENCES UM_ORG_ROLE(UM_ROLE_ID) ON DELETE CASCADE,
+            CONSTRAINT FK_UM_ORG_ROLE_GROUP_UM_ORG_GROUP FOREIGN KEY (UM_GROUP_ID) REFERENCES UM_ORG_GROUP(UM_GROUP_ID) ON DELETE CASCADE
+);
+
+IF NOT  EXISTS (SELECT * FROM SYS.OBJECTS WHERE OBJECT_ID = OBJECT_ID(N'[DBO].[UM_ORG_ROLE_PERMISSION]') AND TYPE IN (N'U'))
+CREATE TABLE UM_ORG_ROLE_PERMISSION(
+            UM_PERMISSION_ID INTEGER NOT NULL,
+            UM_ROLE_ID VARCHAR(255) NOT NULL,
+            CONSTRAINT FK_UM_ORG_ROLE_PERMISSION_UM_ORG_ROLE FOREIGN KEY (UM_ROLE_ID) REFERENCES UM_ORG_ROLE(UM_ROLE_ID) ON DELETE CASCADE,
+            CONSTRAINT FK_UM_ORG_ROLE_PERMISSION_UM_ORG_PERMISSION FOREIGN KEY (UM_PERMISSION_ID) REFERENCES UM_ORG_PERMISSION(UM_ID) ON DELETE CASCADE
+);
+
+IF NOT EXISTS ( SELECT * FROM SYS.OBJECTS WHERE OBJECT_ID = OBJECT_ID(N'[DBO].[UM_ORG_HIERARCHY]') AND TYPE IN (N'U'))
+CREATE TABLE UM_ORG_HIERARCHY (
+            UM_PARENT_ID VARCHAR(255) NOT NULL,
+            UM_ID VARCHAR(255) NOT NULL,
+            DEPTH INTEGER,
+            PRIMARY KEY (UM_PARENT_ID, UM_ID),
+            FOREIGN KEY (UM_PARENT_ID) REFERENCES UM_ORG(UM_ID),
+            FOREIGN KEY (UM_ID) REFERENCES UM_ORG(UM_ID)
+);
+
+CREATE TRIGGER UM_ORG_DELETE_TRIGGER ON UM_ORG
+INSTEAD OF DELETE AS
+BEGIN
+	DELETE FROM UM_ORG_HIERARCHY WHERE UM_PARENT_ID IN (SELECT UM_ID FROM deleted)
+	DELETE FROM UM_ORG_HIERARCHY WHERE UM_ID IN (SELECT UM_ID FROM deleted)
+    DELETE FROM UM_ORG WHERE UM_PARENT_ID IN (SELECT UM_ID FROM deleted)
+	DELETE FROM UM_ORG WHERE UM_ID IN (SELECT UM_ID FROM deleted)
+END;
```

Comment:
is this for cascade deleting?

Reply:
Yes, done through a trigger

Reply:
couldn't see a similar approach taken for tables such as UM_ORG_ATTRIBUTE. any specific reason?

Reply:
also there is an indentation issue

Reply:
This was due to handle issue with having multiple cascade paths in UM_ORG and UM_ORG_HIERARCHY tables upon deletion of a record in UM_ORG.